### PR TITLE
Show instrument descriptions in New Score Wizard

### PR DESCRIPTION
**Pull request by shoogle**
Instrument descriptions are displayed in the bottom left corner of the New Score Wizard dialog. The description text is taken from instruments.xml (i.e. the online instruments spreadsheet).

### Timpani

image

### G Dizi

image

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
This wouled men that those need to get translated too, currently they are not translatable, as far as I can tell

**Comment by shoogle** [comment]
> This would mean that those need to get translated too

Will happen in a future PR

> And what about instruments that don't have a description

Right now every instrument has a description. If a future instrument doesn't have a description then I'm inclined to leave the footer text blank to encourage adding a description.

Current descriptions are not very useful (in most cases it is indeed just the instrument name) but later they will be improved in the [online spreadsheet](https://docs.google.com/spreadsheets/d/1SwqZb8lq5rfv5regPSA10drWjUAoi65EuMoYtG-4k5s/edit#gid=516529997) and updated in a future PR.

File: src/instrumentsscene/view/instrumentlistmodel.cpp [hunk 1]
```diff
@@ -395,6 +397,16 @@ bool InstrumentListModel::hasSelection() const
...
 }
 
+QString InstrumentListModel::selectedInstrumentDescription() const
+{
+    QList<int> selectedRows = m_selection->selectedRows();
+    if (selectedRows.length() != 1) {
+        return QString();
+    }
+    CombinedInstrument instrument = m_instruments.at(selectedRows.at(0));
+    return instrument.templates.at(instrument.currentTemplateIndex)->description;
```
Comment (Jojo-Schmitz) [inline src/instrumentsscene/view/instrumentlistmodel.cpp:407]: I guess this is the place where you'd need to pick on the translation

File: src/project/qml/MuseScore/Project/internal/ChooseInstrumentsAndTemplatesPage.qml [hunk 2]
```diff
@@ -33,6 +33,10 @@ Item {
 
     property string preferredScoreCreationMode: ""
 
+    property string description: pagesStack.currentIndex === 0
+                                 ? instrumentsPage.description
```
Comment (Jojo-Schmitz) [inline src/project/qml/MuseScore/Project/internal/ChooseInstrumentsAndTemplatesPage.qml:37]: and/or here?

File: src/instrumentsscene/view/instrumentlistmodel.cpp (hunk 1 above)
Comment (shoogle) [reply src/instrumentsscene/view/instrumentlistmodel.cpp:407]: Probably

File: src/project/qml/MuseScore/Project/NewScoreDialog.qml [hunk 3]
```diff
@@ -58,9 +58,17 @@ StyledDialogView {
             Layout.fillHeight: true
             Layout.fillWidth: true
 
+            anchors.top: parent.top
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/NewScoreDialog.qml:61]: you shouldn't use anchors for items in Layout. 
please, see https://doc.qt.io/qt-5/qml-qtquick-layouts-columnlayout.html

File: src/project/qml/MuseScore/Project/NewScoreDialog.qml [hunk 4]
```diff
@@ -87,8 +97,32 @@ StyledDialogView {
...
+
+            anchors.top: buttons.top
+            anchors.left: parent.left
+            anchors.right: buttons.left
+            anchors.bottom: buttons.bottom
+            anchors.leftMargin: 12
+            anchors.rightMargin: 12
+
+            opacity: 0.7
+            font.pixelSize: 12
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/NewScoreDialog.qml:114]: please, use one of our fonts from the theme
for example, bodyFont

File: src/project/qml/MuseScore/Project/NewScoreDialog.qml [hunk 5]
```diff
@@ -87,8 +97,32 @@ StyledDialogView {
...
+            lineHeightMode: Text.FixedHeight
+            lineHeight: 14
+            horizontalAlignment: Text.AlignLeft
+            verticalAlignment: Text.AlignVCenter
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/NewScoreDialog.qml:118]: it is the default value

File: src/project/qml/MuseScore/Project/NewScoreDialog.qml [hunk 6]
```diff
@@ -58,9 +58,17 @@ StyledDialogView {
...
+            anchors.top: parent.top
+            anchors.left: parent.left
+            anchors.right: parent.right
+            anchors.bottom: buttons.top
+            anchors.bottomMargin: 24
+
             ChooseInstrumentsAndTemplatesPage {
                 id: chooseInstrumentsAndTemplatePage
 
+                anchors.fill: parent
```
Comment (Eism) [inline src/project/qml/MuseScore/Project/NewScoreDialog.qml:70]: I think it is not necessary

File: src/project/qml/MuseScore/Project/NewScoreDialog.qml (hunk 3 above)
Comment (shoogle) [reply src/project/qml/MuseScore/Project/NewScoreDialog.qml:61]: I think anchors are easier here so I'll just remove the ColumnLayout unless there are any objections?

Comment (Eism) [reply src/project/qml/MuseScore/Project/NewScoreDialog.qml:61]: I disagree, just compare how it looked before and how it looks now

**Review by Eism — APPROVED** [review]

**Pull request merged**